CoreMapView/src/au/gov/asd/tac/constellation/views/mapview2/layers/ThiessenPolygonsLayer.java

```diff
@@ -159,91 +152,80 @@ private void sortNodes(final List<AbstractMarker> markers) {
     private void calculateBisectors() {
         bisectorLines.clear();
 
-        // Set to store markers pairs whose bisectors haev already been calculated
-        final Set<String> calculatedPairs = new HashSet<>();
-
         // Loop through markers
-        for (final Integer id : nodesOnScreen.keySet()) {
-
+        for (int i = 0; i < nodesOnScreen.keySet().toArray().length; i++) {
+            final Integer id = (Integer) nodesOnScreen.keySet().toArray()[i];
             // Loop through markers for marker
-            for (final Integer id2 : nodesOnScreen.keySet()) {
-                // If the markers don't have the same ID value
-                if (id.intValue() != id2.intValue()) {
-                    // Create the pair keys
-                    String idPair = id + "," + id2;
-                    String idPair2 = id2 + "," + id;
+            for (int j = i + 1; j < nodesOnScreen.keySet().toArray().length; j++) {
 
-                    // If the bisector hasn't been calculated yet
-                    if (!calculatedPairs.contains(idPair) && !calculatedPairs.contains(idPair2)) {
-                        // Add the keys to the set
-                        calculatedPairs.add(idPair);
-                        calculatedPairs.add(idPair2);
+                final Integer id2 = (Integer) nodesOnScreen.keySet().toArray()[j];
```

Comment:
wondering if you perhaps need to declare the array in a variable outside of the 2 for loops as since sets aren't ordered, there isn't a way to ensure that the array is ordered the same way every time you call the function